### ISSUE TITLE
Popups should be programmatically determinable

### ISSUE THREAD

**Issue by Malvoz** [Maps4HTML/Web-Map-Custom-Element#354]
The `leaflet-popup` is a plain `<div>` (click any marker to view a popup here: https://maps4html.org/experiments/index.html). I believe it is a failure of SC [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships) to not convey its structure (as visually presented) to <span title="Assistive Technology">ATs</span>. This issue is on the list of [upstream accessibility issues (for popups)](https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/247#popups).

I'd like to think that [`role="dialog"`](https://www.w3.org/TR/wai-aria-1.2/#dialog) is appropriate for the element, but I'm not sure (we can hopefully get some feedback on this per https://github.com/w3c/aria/issues/1421#issuecomment-791645912). WDYT? Is `dialog` appropriate for popups?

/cc @shepazu and @nchan0154, in case you have any opinions on this. 🤗

**Comment by Malvoz** [comment]
Accessibility expert Scott O'Hara stated in https://github.com/w3c/aria/issues/1421#issuecomment-824152117:

> per the provided example, i could see that as a non-modal dialog.

(the example provided was https://maps4html.org/experiments/index.html)

With that response I do think we should use `role="dialog"` and [`aria-modal="false"`](https://www.w3.org/TR/wai-aria-1.2/#aria-modal) for popups. I'm assuming we'll continue to use popups to some extent despite https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/403.

**Comment by nchan0154** [comment]
Apologies for missing this earlier. I agree that `role="dialog"` and `aria-modal="false"` are most appropriate here (feeling validated by this as I had decided this in my head before I saw Scott's comment 😂) In the specific case here I do think the appropriate accessible name is the `name` as the content of the tooltip, and it would also be great if Leaflet let you pass this in as an option.

I'm also noticing this very interesting bug on NVDA + Chrome/Firefox where opening a popup with the keyboard causes the popup to open very left and very up of the actual point, so much so that the map scrolls. I'm not sure how much of this is MapML and how much of it is Leaflet since last I checked, Leafalet didn't even have popups that were keyboard accessible, but possibly worth noting.

**Comment by Malvoz** [comment]
Thanks Nic.

> I do think the appropriate accessible name is the name as the content of the tooltip

I agree, and it'd indeed be great if Leaflet provided that as an option. I'll file an issue on that with them.

> I'm also noticing this very interesting bug on NVDA + Chrome/Firefox where opening a popup with the keyboard causes the popup to open very left and very up of the actual point, so much so that the map scrolls. I'm not sure how much of this is MapML and how much of it is Leaflet since last I checked, Leafalet didn't even have popups that were keyboard accessible, but possibly worth noting.

The bug you're seeing with misplaced popups has been noted in comments before, but it deserves its own issue: filed https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/423. And it's true Leaflet markers aren't operable using screen readers, setting `role="button"` on markers fixes the issue (upstream issue has been filed as noted in https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/247#markers). It has been fixed in this component however.

**Comment by Malvoz** [comment]
I was investigating this a bit yesterday and found that using NVDA <q>dialog</q> is not announced if the same element has `tabindex="-1"` and `role="dialog"`.

With that in mind I believe the requirements are:

1. `leaflet-popup` has `role="dialog"` (this ensures all content of the popup is encapsulated in the `dialog`, including the button that closes it)
2. `leaflet-popup` has `aria-labelledby="{id-of-triggering-marker}"`
3. Move `tabindex="-1"` (currently set on `mapml-popup-content`) to `leaflet-popup-content-wrapper`
